bigframes/ml/forecasting.py

```diff
@@ -25,29 +25,174 @@
+        holiday_region: Optional[str] = None,
+        clean_spikes_and_dips: bool = True,
+        adjust_step_changes: bool = True,
+        time_series_length_fraction: Optional[float] = None,
+        min_time_series_length: Optional[int] = None,
+        max_time_series_length: Optional[int] = None,
+        trend_smoothing_window_size: Optional[int] = None,
+        decompose_time_series: bool = True,
+    ):
+        self.horizon = horizon
+        self.auto_arima = auto_arima
+        self.auto_arima_max_order = auto_arima_max_order
+        self.auto_arima_min_order = auto_arima_min_order
+        self.data_frequency = data_frequency
+        self.include_drift = include_drift
+        self.holiday_region = holiday_region
+        self.clean_spikes_and_dips = clean_spikes_and_dips
+        self.adjust_step_changes = adjust_step_changes
+        self.time_series_length_fraction = time_series_length_fraction
+        self.min_time_series_length = min_time_series_length
+        self.max_time_series_length = max_time_series_length
+        self.trend_smoothing_window_size = trend_smoothing_window_size
+        self.decompose_time_series = decompose_time_series
+        # TODO(garrettwu) add order and seasonalities params, which need struct/array
 
-    def __init__(self):
         self._bqml_model: Optional[core.BqmlModel] = None
         self._bqml_model_factory = globals.bqml_model_factory()
 
     @classmethod
     def _from_bq(cls, session: bigframes.Session, model: bigquery.Model) -> ARIMAPlus:
         assert model.model_type == "ARIMA_PLUS"
 
-        kwargs: Dict[str, str | int | bool | float | List[str]] = {}
+        kwargs: dict = {}
+        last_fitting = model.training_runs[-1]["trainingOptions"]
+
+        dummy_arima = cls()
+        for bf_param, bf_value in dummy_arima.__dict__.items():
+            bqml_param = _BQML_PARAMS_MAPPING.get(bf_param)
+            if bqml_param in last_fitting:
+                # Convert types
+                if bf_param in ["time_series_length_fraction"]:
+                    kwargs[bf_param] = float(last_fitting[bqml_param])
+                elif bf_param in [
+                    "auto_arima_max_order",
+                    "auto_arima_min_order",
+                    "min_time_series_length",
+                    "max_time_series_length",
+                    "trend_smoothing_window_size",
+                ]:
+                    kwargs[bf_param] = int(last_fitting[bqml_param])
+                elif bf_param in ["holiday_region"]:
+                    kwargs[bf_param] = str(last_fitting[bqml_param])
+                else:
+                    kwargs[bf_param] = type(bf_value)(last_fitting[bqml_param])
 
         new_arima_plus = cls(**kwargs)
         new_arima_plus._bqml_model = core.BqmlModel(session, model)
         return new_arima_plus
 
     @property
-    def _bqml_options(self) -> Dict[str, str | int | bool | float | List[str]]:
+    def _bqml_options(self) -> dict:
```

Comment:
[nit] `Dict` for consistency

Reply:
I try to use build-in type annotations (list, dict, etc.) by default. Which are neater since don't require imports.